charset_normalizer/md.py

```diff
@@ -495,18 +461,17 @@ def is_suspiciously_successive_range(
             return False
 
     # Japanese Exception
-    if unicode_range_a in ["Katakana", "Hiragana"] and unicode_range_b in [
-        "Katakana",
-        "Hiragana",
-    ]:
-        return False
-
     if unicode_range_a in ["Katakana", "Hiragana"] or unicode_range_b in [
         "Katakana",
         "Hiragana",
     ]:
         if "CJK" in unicode_range_a or "CJK" in unicode_range_b:
             return False
+        if unicode_range_a in ["Katakana", "Hiragana"] and unicode_range_b in [
+            "Katakana",
+            "Hiragana",
+        ]:
```

Comment:
Isn't this check duplicated 3 lines above?

Reply:
It's `and` vs `or`, but overall it could be made more efficient I guess.

Reply:
Well, you could hoist the values of `unicode_range_a in ["Katakana", "Hiragana"]` and `unicode_range_b in ["Katakana", "Hiragana"]` into local variables, then simplify this to

```
if range_a_katahira or range_b_katahira:
  if "CJK" ...:
  if range_a_katahira and range_b_katahira:
     ...
```

Reply:
Yes, good idea